### PR TITLE
Payment page riddl definition (#29)

### PR DESCRIPTION
**Pull request by JamesYopp**
Payment page RIDDL definition

### CODE REVIEW

**Review by reid-spencer — CHANGES_REQUESTED** [review]

File: design/src/main/riddl/application/pages/payment.riddl [hunk 1]
```diff
@@ -1 +1,39 @@
-page PaymentPage {???}
\ No newline at end of file
+record PaymentInfo is {
+  nameOnCard: String,
+  cardNumber: String,
```
[the remark this answers is not in the document]
Comment (JamesYopp) [reply design/src/main/riddl/application/pages/payment.riddl:3]: I am hoping to demonstrate how we would manage PCI compliance. The easiest way to do this is to delegate the PaymentInfo form to the credit card processor. What we should persist on our end is a credit card token. We will not know the pattern or length of this token as it varies from processor to processor.

File: design/src/main/riddl/application/pages/payment.riddl [hunk 2]
```diff
@@ -1 +1,39 @@
...
+  billingAddress: ksoTemplateAppDomain.Address,
+  savePaymentInfoToProfile: Boolean,
+  copyPaymentAddressFromBillingAddress: Boolean
+} described as "Credit Card info to make payment. CVV CANNOT be stored. Ideally, the card number would be a token provided by the credit card processor."
+
+type NameOnCard: String
+type CardNumber: String briefly "Ideally, the card number would be a token provided by the credit card processor"
+type ExpirationDate: Date
+type SavePaymentInfoToProfile: Boolean
+type CopyPaymentInfoFromBillingInfo: Boolean
```
[the remark this answers is not in the document]
Comment (JamesYopp) [reply design/src/main/riddl/application/pages/payment.riddl:14]: The issue I ran into is I needed a type later when I was defining inputs for the form. I could not say:

`selector savePaymentInfoToProfile accepts Boolean`

This results in the following error:
`[error] Error: application/pages/payment.riddl(34:47):
Path 'Boolean' was not resolved, in selector 'savePaymentInfoToProfile'
and it should refer to a Type:
    selector savePaymentInfoToProfile accepts Boolean
                                              ^`

File: design/src/main/riddl/ksoapp.md [hunk 3]
```diff
@@ -56,4 +56,6 @@ The basic function and flow of the application is as follows:
   - blog post on "Eventual Consistency"
   - blog post on "CQRS"
 
-Should we consider presenting the applicaiton in English and French?
+TODO: We need to present the applicaiton in English and French
+TODO: Demonstrate GDPR compliance (right to be forgotten)
+TODO: Demonstrate PII best practices -- encrypt PII for storage
```
[the remark this answers is not in the document]
Comment (JamesYopp) [reply design/src/main/riddl/ksoapp.md:61]: These comments are in a *.md file. Why would they not compile?

**Comment by JamesYopp** [comment]
All changes were included in a separate PR

**Pull request closed without merging**